### DISCUSSION TITLE
What about quantum-safe algorithms?

### DISCUSSION THREAD

**Discussion by dzobbe** [gramineproject/gramine#1748] [General]
Hi all,

This discussion I am opening is really just to hear your plan (if any) with regards to post-quantum cryptography.

I have seen [this](https://github.com/open-quantum-safe/oqs-provider/) interesting project. Are you waiting for the standardization process to be completed? NIST selected Kyber for  pubkey encryption. 

Thanks in advance and I take the opportunity to congrat with you because your runtime is really stable and well performing.

Cheers!!

**Comment by mkow** [comment]
> This discussion I am opening is really just to hear your plan (if any) with regards to post-quantum cryptography.

We (or at least I) don't plan anything in this direction, there are many reasons, including:
- It's still very futuristic and rather a research topic.
- We don't even know if quantum computers of considerable size are actually possible to build, and even if, that probably won't happen too soon.
- SGX is inherently insecure against quantum computers, because it uses RSA, we'd need to switch to a different technology. Just for the record, I'm not blaming Intel here for this choice, AFAIR post-quantum crypto isn't too performant, requires huge keys, etc., and the risk seems very remote. And I think all other modern TEEs made the same choice, so we have no other options.

> I have seen [this](https://github.com/open-quantum-safe/oqs-provider/) interesting project. Are you waiting for the standardization process to be completed? NIST selected Kyber for pubkey encryption.

See above.

> Thanks in advance and I take the opportunity to congrat with you because your runtime is really stable and well performing.

Nice to hear that!

**Comment by monavij** [comment]
> * It's still very futuristic and rather a research topic.
> * We don't even know if quantum computers of considerable size are actually possible to build, and even if, that probably won't happen too soon.
> * SGX is inherently insecure against quantum computers, because it uses RSA, we'd need to switch to a different technology. Just for the record, I'm not blaming Intel here for this choice, AFAIR post-quantum crypto isn't too performant, requires huge keys, etc., and the risk seems very remote. And I think all other modern TEEs made the same choice, so we have no other options.
> 
While Michal is correct about #1 and #2, let me just correct his statement about SGX. Intel’s practice is to align our products and features to PQC best practices.  We are monitoring NIST’s recommendations and plan to update SGX in the future for PQC as NIST’s guidance evolves, including updates or changes to RSA.
